hyperlane/relayer_runner.go

```diff
@@ -83,47 +84,38 @@ func (r *RelayerRunner) Run(ctx context.Context) error {
 					continue
 				}
 
-				isTransferValid, err := r.validateHyperlaneTransfer(ctx, transfer)
+				destinationTxHash, destinationChainID, err := r.relayHandler.Relay(ctx, transfer.SourceChainID, transfer.MessageSentTx)
 				if err != nil {
-					lmt.Logger(ctx).Warn(
-						"failed to validate hyperlane transfer",
-						zap.Int64("transferId", transfer.ID),
-						zap.String("txHash", transfer.MessageSentTx),
-						zap.Error(err),
-					)
-					continue
-				}
-
-				if !isTransferValid {
-					lmt.Logger(ctx).Warn(
-						"abandoning invalid hyperlane transfer",
-						zap.Int64("transferId", transfer.ID),
-						zap.String("txHash", transfer.MessageSentTx),
-						zap.Error(err),
-					)
-
-					if _, err := r.db.SetMessageStatus(ctx, db.SetMessageStatusParams{
-						TransferStatus:     dbtypes.TransferStatusAbandoned,
-						SourceChainID:      transfer.SourceChainID,
-						DestinationChainID: transfer.DestinationChainID,
-						MessageID:          transfer.MessageID,
-					}); err != nil {
-						lmt.Logger(ctx).Error(
-							"error updating invalid transfer status",
+					// Unrecoverable error
+					if strings.Contains("execution reverted", err.Error()) {
+						lmt.Logger(ctx).Warn(
+							"abandoning hyperlane transfer",
 							zap.Int64("transferId", transfer.ID),
 							zap.String("txHash", transfer.MessageSentTx),
 							zap.Error(err),
 						)
+
+						if _, err := r.db.SetMessageStatus(ctx, db.SetMessageStatusParams{
+							TransferStatus:     dbtypes.TransferStatusAbandoned,
```

Comment:
can you also fill the `TransferStatusMessage` field here with the revert message? bonus if you also rename this query method to `SetTransferStatus`